### PR TITLE
Animation 时支持自定义贝塞尔曲线 timingFunction

### PR DESCRIPTION
**Pull request by zcfan**
之前提过一个 issue，周末有空自己看了下 https://github.com/Tencent/Hippy/issues/632

这是个 prove of concept，仅对 iOS 做了简单的验证，效果如视频，Android 要晚些才有空检查实现是不是正确。

https://user-images.githubusercontent.com/1666185/120099137-8233e580-c16c-11eb-812f-8238d8fa7bc9.mov

timingFunction 如下：

```javascript
this.horizonAnimation = new Animation({
  startValue: 150,
  toValue: 20,
  duration: 1000,
  delay: 500,
  mode: 'timing',
  timingFunction: 'cubic-bezier(.64,1.82,.67,.71)',  // <------------- spring 效果的自定义贝塞尔曲线
  repeatCount: 'loop',
});
```

新属性叫 cubic-bezier 有两个考虑：

1. 不与之前的 `ease_bezier` 冲突，现有项目可能依赖之前的表现（话说之前的 ease_bezier 为什么用下划线，不和其他的 ease-in 之类的一致用中划线？有特别考虑还是单纯手滑打错了？😂）
2. 符合 web 开发中对 easing- function 的定义，详见：https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function

请初步评估一下接口设计是否合理，可行的话我再完善实现，补充缺失的 web 端

### CODE REVIEW

File: ios/sdk/module/animation/HippyExtAnimation.m [hunk 1]
```diff
@@ -239,12 +236,32 @@ - (CAAnimation *)animationOfView:(UIView *)view forProp:(NSString *)prop {
...
 }
 
+- (nullable CAMediaTimingFunction *)makeCustomBezierFunction:(NSString *)timingFunction {
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"^cubic-bezier\\(([+-]?(?:[0-9]*[.])?[0-9]+),([+-]?(?:[0-9]*[.])?[0-9]+),([+-]?(?:[0-9]*[.])?[0-9]+),([+-]?(?:[0-9]*[.])?[0-9]+)\\)$" options:NSRegularExpressionCaseInsensitive error:nil];
+    if (!regex) return nil;
+    NSRange searchedRange = NSMakeRange(0, [timingFunction length]);
+    NSTextCheckingResult *match = [regex matchesInString:timingFunction options:0 range: searchedRange][0];
+    NSString *matchText = [timingFunction substringWithRange:[match range]];
+    float (^getValue)(NSUInteger) = ^(NSUInteger index) {
+        NSRange range = [match rangeAtIndex: index];
```
Comment (ozonelmy) [inline ios/sdk/module/animation/HippyExtAnimation.m:254]: 这个方法是否存在越界可能

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=785) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/Hippy?pullRequest=785) before we can accept your contribution.<br/>**3** out of **4** committers have signed the CLA.<br/><br/>:white_check_mark: zcfan<br/>:white_check_mark: zoomchan-cxj<br/>:white_check_mark: ozonelmy<br/>:x: cfanzhang<br/><hr/>**cfanzhang** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/Hippy?pullRequest=785) it.</sub>

**Comment by zcfan** [comment]
补充了 @hippy/react-web 的支持和文档，验证了 android 上的表现，辛苦 review 下

**Comment by zcfan** [comment]
https://user-images.githubusercontent.com/1666185/129865234-0ffa5374-e261-4c7e-ba4b-ac9edf9ed110.mov

补充 demo

File: docs/hippy-react/best-practices.md [hunk 2]
```diff
@@ -151,7 +151,7 @@ Animation支持的动画配置包括：
   * “ease-in”：使用加速插值器，动画速度将随时间逐渐增加；
   * “ease-out”：使用减速插值器，动画速度将随时间逐渐减小；
   * “ease-in-out”：使用加减速插值器，动画速度前半段先随时间逐渐增加，后半段速度将逐渐减小；
-  * “ease_bezier”：暂不支持；
+  * “cubic-bezeir”：使用自定义贝塞尔曲线，与 [css transition-timing-function 的 cubic-bezier](https://developer.mozilla.org/en-US/docs/Web/CSS/transition-timing-function) 一致；
```
Comment (zoomchan-cxj) [inline docs/hippy-react/best-practices.md:154]: cubic-bezeir 拼写错误

Comment (zoomchan-cxj) [inline docs/hippy-react/best-practices.md:154]: 补充：`最低支持版本 2.9.0`

File: docs/hippy-react/modules.md [hunk 3]
```diff
@@ -40,7 +40,7 @@
   * `ease-in`：使用加速插值器，动画速度将随时间逐渐增加；
   * `ease-out`：使用减速插值器，动画速度将随时间逐渐减小；
   * `ease-in-out`：使用加减速插值器，动画速度前半段先随时间逐渐增加，后半段速度将逐渐减小；
-  * `ease_bezier`：暂不支持；
+  * `cubic-bezeir`：使用自定义贝塞尔曲线，与 [css transition-timing-function 的 cubic-bezier](https://developer.mozilla.org/en-US/docs/Web/CSS/transition-timing-function) 一致；
```
Comment (zoomchan-cxj) [inline docs/hippy-react/modules.md:43]: cubic-bezeir 拼写错误，补充最低支持版本

**Review by siguangli — APPROVED** [review]

**Pull request merged**